### PR TITLE
SAN-3726; Better Login Error Messages

### PR DESCRIPTION
**Pull request by rsandor**
This PR addresses two issues:
1. Returns specific messages as given by GitHub when there is a login error
2. Fixes rejection tests (there is no `promise, class, regex` signature for `.isRejected`)

Reviewers:
- [x] @bkendall

### CODE REVIEW

File: test/unit/login.js [hunk 1]
```diff
@@ -49,8 +49,7 @@ describe('Login', () => {
         Login._read.onFirstCall().rejects(new Error('robot'))
         return assert.isRejected(
           Login.login(mockArgs),
-          Error,
-          /robot/
```
Comment (bkendall) [inline test/unit/login.js:53]: I think having this message check here is still correct. This prevents something like "cannot call _something_ of undefined" from coming back as the rejection error and the test still passing.

File: test/unit/login.js [hunk 2]
```diff
@@ -202,7 +198,6 @@ describe('Login', () => {
         request.post.yieldsAsync(new Error('robot'))
         return assert.isRejected(
           Login._makeRequest(mockCreds),
-          Error,
```
Comment (bkendall) [inline test/unit/login.js:205]: you removed the `Error` from these?

**Comment by bkendall** [comment]
> Fixes rejection tests (there is no promise, class, regex signature for .isRejected)

I see. now my comments are not quite right. I do see a `(promise, class, string)` signature though - maybe we should be using that instead.

**Comment by rsandor** [comment]
@bkendall - I couldn't get those tests to pass correctly for the life of me. But I will give it another try, perhaps I was just braindead that day :dancers:.

**Comment by bkendall** [comment]
nice!

**Pull request merged**